lib/Graph/Nodes.cpp

```diff
@@ -1463,6 +1463,15 @@ bool BatchedReduceMinNode::verify() const {
   return isValid;
 }
 
+bool BatchedReduceMaxNode::verify() const {
+  bool isValid = checkType(getResult(), getBatch().getElementType(), this);
+
+  isValid &=
+      expectCompareTrue("Invalid shape", getBatch().dims().size(), size_t(0),
+                        this, CompareOperatorGreaterThan<size_t>());
+  return isValid;
+}
+
```

Comment:
This is an exact duplicate of verify for `BatchedReduceMeanNode` and `BatchedReduceMinNode`. Can you colocate all of the logic here? Could be done with a simple helper, e.g. see `verifyArithmetic()`.

Reply:
Common verification code for `BatchedReduceMeanNode`, `BatchedReduceMinNode`, `BatchedReduceMaxNode` and `BatchedReduceAddNode`.